pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2_ddl_handler.go

```diff
@@ -0,0 +1,388 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package priorityqueue
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/pkg/ddl/notifier"
+	"github.com/pingcap/tidb/pkg/infoschema"
+	"github.com/pingcap/tidb/pkg/meta/model"
+	"github.com/pingcap/tidb/pkg/sessionctx"
+	"github.com/pingcap/tidb/pkg/sessionctx/variable"
+	"github.com/pingcap/tidb/pkg/statistics"
+	"github.com/pingcap/tidb/pkg/statistics/handle/autoanalyze/exec"
+	statslogutil "github.com/pingcap/tidb/pkg/statistics/handle/logutil"
+	"go.uber.org/zap"
+)
+
+// HandleDDLEvent handles DDL events for the priority queue.
+func (pq *AnalysisPriorityQueueV2) HandleDDLEvent(_ context.Context, sctx sessionctx.Context, event *notifier.SchemaChangeEvent) (err error) {
+	pq.syncFields.mu.Lock()
+	defer pq.syncFields.mu.Unlock()
+	// If the priority queue is not initialized, we should retry later.
+	if !pq.syncFields.initialized {
+		return notifier.ErrNotReadyRetryLater
+	}
+
+	defer func() {
+		if err != nil {
+			actionType := event.GetType().String()
+			statslogutil.StatsLogger().Error(fmt.Sprintf("Failed to handle %s event", actionType),
+				zap.Error(err),
+				zap.String("event", event.String()),
+			)
+		}
+	}()
+
+	switch event.GetType() {
+	case model.ActionAddIndex:
+		err = pq.handleAddIndexEvent(sctx, event)
+	case model.ActionTruncateTable:
+		err = pq.handleTruncateTableEvent(sctx, event)
+	case model.ActionDropTable:
+		err = pq.handleDropTableEvent(sctx, event)
+	case model.ActionTruncateTablePartition:
+		err = pq.handleTruncateTablePartitionEvent(sctx, event)
+	case model.ActionDropTablePartition:
+		err = pq.handleDropTablePartitionEvent(sctx, event)
+	case model.ActionExchangeTablePartition:
+		err = pq.handleExchangeTablePartitionEvent(sctx, event)
+	case model.ActionReorganizePartition:
+		err = pq.handleReorganizePartitionEvent(sctx, event)
+	case model.ActionAlterTablePartitioning:
+		err = pq.handleAlterTablePartitioningEvent(sctx, event)
+	case model.ActionRemovePartitioning:
+		err = pq.handleRemovePartitioningEvent(sctx, event)
+	default:
+		// Ignore other DDL events.
```

Comment:
Shall we consider print a log in the default branch

Reply:
It is OK for now.